### PR TITLE
(GH-523) Use .Net Global tool version of KuduSync

### LINKED ISSUE #523: When building .Net Core projects prefer .Net Global Tool when available
At the minute, even though building a .Net Core project, some of the tools that are used are the full .Net Framework version.  When building a .net core project we can assume that usage of .Net Global Tools are available, so we should prefer these when available.

- [x] GitVersion
- [x] GitReleaseManager
- [x] Codecov
- [x] Coveralls
- [x] Wyam
- [ ] KuduSync
- [x] ReportGenerator

### PR DESCRIPTION
**Pull request by gep13**
This is the last of the tools that can be moved to .net core.

Fixes #523

### CODE REVIEW

File: Cake.Recipe/Content/toolsettings.cake [hunk 1]
```diff
@@ -57,7 +58,9 @@ public static class ToolSettings
         string gitReleaseManagerGlobalTool = "#tool dotnet:?package=GitReleaseManager.Tool&version=0.11.0",
         string gitVersionGlobalTool = "#tool dotnet:?package=GitVersion.Tool&version=5.3.4",
         string reportGeneratorGlobalTool = "#tool dotnet:?package=dotnet-reportgenerator-globaltool&version=4.6.1",
-        string wyamGlobalTool = "#tool dotnet:?package=Wyam.Tool&version=2.2.9"
+        string wyamGlobalTool = "#tool dotnet:?package=Wyam.Tool&version=2.2.9",
+        // This is using an unofficial build of kudusync so that we can have a .Net Global tool version.  This was generated from this PR: https://github.com/projectkudu/KuduSync.NET/pull/27
+        string kuduSyncGlobalTool = "#tool dotnet:https://www.myget.org/F/cake-contrib/api/v3/index.json?package=KuduSync.Tool&version=1.5.4-gc5cc5a2a19"
```
Comment (AdmiringWorm) [inline Cake.Recipe/Content/toolsettings.cake:63]: Just reiterating what I said on gitter (in a different format :smile:):

> I am a bit uncertain about using a myget.org URL for the unofficial build, mainly due to retention policies that may cause that specific version to suddenly disappear.

Comment (gep13) [reply Cake.Recipe/Content/toolsettings.cake:63]: This package version is pinned, so it shouldn’t be cleaned up by retention policies, so we should be able to take this dependency without any concerns. The KuduSync tool isn’t one that has changed recently, so I don’t foresee us pushing multiple package versions.

**Comment by gep13** [comment]
@AdmiringWorm I spoke with @devlead on Slack last night, and he was fine with this getting merged.  I will move him as a reviewer.

**Comment by AdmiringWorm** [comment]
@gep13 your changes have been merged, thanks for your contribution 👍

and thank you @devlead for making it possible

**Pull request merged**